inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -317,11 +318,30 @@ CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>&
     return OK;
 }
 
+namespace {
+    std::vector<std::shared_ptr<ngraph::Function>> get_all_sub_functions(const std::shared_ptr<ngraph::Function>& func) {
+        std::vector<std::shared_ptr<ngraph::Function>> sub_functions;
+        auto ops = func->get_ops();
+        size_t ops_size = ops.size();
+        for (size_t i = 0; i < ops_size; ++i) {
+            if (dynamic_cast<ngraph::op::util::SubGraphOp*>(ops[i].get()) != nullptr) {
+                auto sub_func = dynamic_cast<ngraph::op::util::SubGraphOp*>(ops[i].get())->get_function();
+                const auto& inner_func_ops = sub_func->get_ops();
+                ops.insert(ops.end(), inner_func_ops.begin(), inner_func_ops.end());
+                ops_size += inner_func_ops.size();
+                sub_functions.push_back(sub_func);
+            }
+        }
+        return sub_functions;
+    }
+} // namespace
+
 void
 CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>& inputShapes) {
     OV_ITT_SCOPED_TASK(itt::domains::IE, "CNNNetworkNGraphImpl::reshape");
 
     auto params = _ngraph_function->get_parameters();
+    auto sub_functions = get_all_sub_functions(_ngraph_function);
```

Comment:
I am not sure if it is an optimum solution. Maybe in order to avoid checking all nodes during reshape `Function` should have field `m_contain_sub_functions`? Or maybe there is some other possibility to replace parameters in sub-functions (SubGraphOp's bodies)